### ISSUE TITLE
Support for InfluxDB/DataDog

### ISSUE THREAD

**Issue by NeQuissimus** [zio/zio-metrics#19]
It would be great to have integration with the TICK stack (and compatible services such as DataDog).
https://github.com/gphat/censorinus could be used for this

**Comment by toxicafunk** [comment]
It seems Telegraf already has a [Prometheus Input Plugin](https://github.com/influxdata/telegraf/blob/release-1.13/plugins/inputs/prometheus/README.md) which means all you need to do is start the [Http Exporter](https://zio.github.io/zio-metrics/docs/essentials/essentials_files#exporters) and configure the endpoint on the plugin:

```
# Read metrics from one or many prometheus clients
[[inputs.prometheus]]
  ## An array of urls to scrape metrics from.
  urls = ["http://localhost:9100/metrics"]
```

**Comment by NeQuissimus** [comment]
Sure but then I need to either run Prometheus infrastructure or change my Telegraf config every time I have a new app instance :)

**Comment by toxicafunk** [comment]
I may be misunderstanding something, the way I see this, you can use zio-metrics-prometheus to instrument your app and then start the Http Exporter. Then you add the Prometheus plugin to your Telegraf and configure the endpoint from the exporter. This would be only one more input source to your existing Telegraf.

Regardless of the above, is there a Telegraf API that can be invoked directly?

**Comment by NeQuissimus** [comment]
Let's say I have my app running in Kubernetes. At any given time I may have 60 instances, depending on how well the app works, those pods restart at will, each time having a new IP.
The way I understand it, the app becomes the Prometheus client, does it not? So Telegraf would have to have a list of all my pod IPs at all times.

If you use censorinus (linked above), you just send simple UDP packets to Telegraf and it forwards those into an InfluxDB, where you can hook up a UI.
There is no fetching of metrics at all, everything flows in one direction (App -> Telegraf -> InfluxDB -> UI), as opposed to the HTTP server Prometheus needs, which is scraped. (and then leads to a App <- Telegraf -> InfluxDB graph)

**Comment by toxicafunk** [comment]
Oh, I get it now! Just one question, the setup you described needs the
Telegraf Statsd input plugin, right?

I'll start work on adding this today.

**Comment by NeQuissimus** [comment]
Yes, it will :+1:

**Comment by toxicafunk** [comment]
Implemented in v0.1.0
